source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -623,6 +641,17 @@ Network::FilterStatus Filter::onNewConnection() {
     connection_duration_timer_->enableTimer(config_->maxDownstreamConnectionDuration().value());
   }
 
+  if (config_->accessLogFlushInterval().has_value()) {
+    access_log_flush_timer_ = read_callbacks_->connection().dispatcher().createTimer(
```

Comment:
Just a quick question. Will the timer and `tcp_proxy` filter run on the same thread? I am thinking about concurrency when accessing `stream_info`. Timer's callback will access `stream_info` to create access log and at the same time `tcp_proxy` may access `stream_info` to update it. Maybe dispatcher assures proper access.

Reply:
yeah.